webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/policies/RuleTarget.scala

```diff
@@ -295,17 +295,20 @@ object RuleTarget extends Loggable {
       targets:          Set[RuleTarget],
       allNodes:         Map[NodeId, Boolean /* isPolicyServer */ ],
       groups:           Map[NodeGroupId, Chunk[NodeId]],
-      allNodesAreThere: Boolean = true // if we are working on a subset of node, set to false
   ): Chunk[NodeId] = {
-    val result = getNodeIdsChunkRec(Chunk.fromIterable(targets), allNodes, groups, allNodesAreThere)
-    result.distinct
+    val result = getNodeIdsChunkRec(Chunk.fromIterable(targets), allNodes.toMap, groups)
```

Comment:
i don't understand the toMap here; allNodes is already a map

Reply:
it's a mapview (or is in latter release) and it will be a noop if the underlying instance is a map without pending operation